plugins/autostart/src/lib.rs

```diff
@@ -46,6 +54,32 @@ impl Serialize for Error {
     }
 }
 
+impl Default for PreferedName {
+    fn default() -> Self {
+        PreferedName::PackageInfoName
+    }
+}
+
+impl PreferedName {
+    pub fn get_value<R: Runtime>(&self, app: &AppHandle<R>) -> String {
+        match self {
+            PreferedName::PackageInfoName => {
+                app.package_info().name.to_string()
+            },
+            PreferedName::PackageInfoCrateName => {
+                app.package_info().crate_name.to_string()
+            },
+            PreferedName::ConfigIdentifier => {
+                app.config().identifier.to_string()
+            },
+            PreferedName::ConfigProductName => {
+                app.config().product_name.clone().unwrap_or("TauriAPP".to_string())
+            },
+            PreferedName::Custom(s) => s.clone(),
+        }
+    }
+}
```

Comment:
We can implement a `From` trait for the builder to be more ergonomic

```suggestion
impl PreferedName {
    pub fn get_value<R: Runtime>(&self, app: &AppHandle<R>) -> String {
        match self {
            PreferedName::PackageInfoName => {
                app.package_info().name.to_string()
            },
            PreferedName::PackageInfoCrateName => {
                app.package_info().crate_name.to_string()
            },
            PreferedName::ConfigIdentifier => {
                app.config().identifier.to_string()
            },
            PreferedName::ConfigProductName => {
                app.config().product_name.clone().unwrap_or("TauriAPP".to_string())
            },
            PreferedName::Custom(s) => s.clone(),
        }
    }
}

impl<S: Into<String>> From<S> for PreferedName {
    fn from(name: S) -> Self {
        PreferedName::Custom(name.into())
    }
}
```